BLE Indoor Positioning/src/main/java/com/nexenio/bleindoorpositioning/IndoorPositioning.java

```diff
@@ -77,31 +82,36 @@ public void onBeaconUpdated(Beacon beacon) {
     private void updateLocation() {
         List<Beacon> usableBeacons = getUsableBeacons(BeaconManager.getInstance().getBeaconMap().values());
 
-        if (usableBeacons.size() < 3) {
+        if (usableBeacons.size() < MINIMUM_BEACON_COUNT) {
             return; // multilateration requires at least 3 beacons
-        } else if (usableBeacons.size() > 3) {
+        } else if (usableBeacons.size() > MINIMUM_BEACON_COUNT) {
             Collections.sort(usableBeacons, Beacon.RssiComparator);
-            Collections.reverse(usableBeacons);
-            for (int beaconIndex = usableBeacons.size() - 1; beaconIndex >= 3; beaconIndex--) {
+            int firstRemovableBeaconIndex = MAXIMUM_BEACON_COUNT;
+            for (int beaconIndex = MINIMUM_BEACON_COUNT; beaconIndex < MAXIMUM_BEACON_COUNT; beaconIndex++) {
                 if (usableBeacons.get(beaconIndex).getFilteredRssi() < minimumRssiThreshold) {
-                    usableBeacons.remove(beaconIndex);
+                    firstRemovableBeaconIndex = beaconIndex;
+                    break;
                 }
             }
+            usableBeacons.subList(firstRemovableBeaconIndex, usableBeacons.size()).clear();
         }
 
         Multilateration multilateration = new Multilateration(usableBeacons);
-        Location location = multilateration.getLocation();
-
-        // The root mean square of multilateration is used to filter out inaccurate locations.
-        // Adjust value to allow location updates with higher deviation
-        if (multilateration.getRMS() < rootMeanSquareThreshold) {
-            locationPredictor.addLocation(location);
-            Location meanLocation = getMeanLocation(2, TimeUnit.SECONDS);
-            if (meanLocation != null) {
-                onLocationUpdated(meanLocation);
+        try {
+            Location location = multilateration.getLocation();
+
+            // The root mean square of multilateration is used to filter out inaccurate locations.
+            // Adjust value to allow location updates with higher deviation
+            if (multilateration.getRMS() < rootMeanSquareThreshold) {
+                locationPredictor.addLocation(location);
+                Location meanLocation = getMeanLocation(2, TimeUnit.SECONDS);
```

Comment:
Could you please introduce a constant here...

Reply:
I wouldn't consider this a magic number. It doesn't cause any confusion when read and is not used multiple times. I would argue that adding a constant for things like these makes the code less readable because of verbosity.

What's your opinion about this, @marvinmirtschin?

Reply:
I'd agree with Steppschuh that sometimes it is more readable when numbers are used, especially in this case as the time unit makes it clear.

Reply:
Sorry for being meddlesome, but without adding a variable for this how we suppose to change the value and test it with another value ?

What if we have a kid running and i want to decrease the time to 1 sec instead of 2?

Reply:
I removed the `meanLocation` in 6e0c92bf519ade28905873c61b4713fd3428db7b and instead  passed the raw location. If the library user wants to use a filtered location, `IndoorPositioning.getMeanLocation(...)` would be the way to go.

Thanks for your input 👍